consul-lambda-registrator/extension/extension.go

```diff
@@ -179,6 +189,10 @@ func (ext *Extension) getExtensionData(ctx context.Context) error {
 
 	// If the extension data has changed then update the cached copy.
 	if !extData.Equals(ext.data) {
+		if ext.dataInit {
```

Comment:
Now that the locking occurs after fetching data from param store, there is a theoretical edge case where we could fetch multiple times from param store simultaneously. I don't think we will run into it since we update param store data so infrequently though.